docs/content/dev/notes.md

```diff
@@ -0,0 +1,50 @@
+# Notes
+
+Each note in HedgeDoc 2 contains the following information:
+
+- publicId (`b604x5885k9k01bq7tsmawvnp0`)
+- a list of aliases (`[hedgedoc-2, hedgedoc-next]`)
+- groupPermissions
+- userPermissions
+- viewCount (`0`)
+- owner
+- revisions
+- authorColors
+- historyEntries
+- description (`All you never wanted to know about notes`)
+- title (`Notes`)
+- tags (`[features, cool, update]`)
+
+The `publicId` is the default possibility of identifying a note. It will be a randomly generated 128-bit value encoded with [base32-encode](https://www.npmjs.com/package/base32-encode) using the crockford variant and converted to lowercase. We use lowercase, because we want to minimize case confusion.
+
+`aliases` are the other way of identifying a note. There can be any number of them and the owner of the note is able to add or remove them. All aliases are just strings (especially to accommodate the old identifier from HedgeDoc 1 [see below](#conversion-of-hedgedoc-1-notes)), but new aliases added with HedgeDoc 2 will only allow the following characters `[a-z0-9\-_]`. This is done to once again prevent case confusion.
+
+`groupPermissions` and `userPermissions` each hold a list of the approriate permissions. Each of those holds a refernce to a note and a user/group and specify what the the user/group is allowed to do. Each of the permissions is additive, that means a user that has only the right to read a note via a group, but the right to write via a different group or directly for his user, is able to write in the note.
+
+The `viewCount` is a simple counter that holds how often the note in question was requested.
+
+`onwer` is the user that created the note or later got ownership of the note. The current owner is able to change the owner of the note to someone else.
+The `revisions` hold all revisions of the note. These are the changes to the note content and by which they were performed.
+
+The `authorColors` each specifiy for the tuple user and note which color should be used to highlight them.
+
+The `historyEntries` hold the history entries this note is refernced in.They are mainly here for the purpose of deleting the history entries on note deletion.
+
+`description`, `tags` and `title` are each information specified in the frontmater of the note. They are extracted and saved in the database to allow the history page to show them and do a search for tags without having to do a full-text search or having to parse the tags of each note on search.
+While `description` and `tags` are only specified by the frontmatter, the title is
+- the content of the *title* field of the frontmatter of the note
+- **OR** the content of the *title* field in the *opengraph* field of the frontmatter of the note
+- **OR** the first level 1 heading of the note
+  which ever of these is the first to not be unspecified.
```

Comment:
Do we, and if so how, handle uploaded notes that do not provide these fields separatel? Or might just be empty in general?

I think we should explicitly declare who handles the extraction of metadata from the note here.

Reply:
The idea at one point was to let the frontend handle this extraction as the code will live in the same space as the backend. So until we try that and fail in accomplishing that I guess that's how.

Reply:
Either way, it should be defined where it should happen. Just to make sure there is no confusion. If we change that decision at some point we can update the docs, but it should be documented in first place to make sure that people who want to develop a client know where whether they have to care about it or not. As well as people debugging it, know where to look for the issue.

Reply:
added

Reply:
So if I got the addition correct, the backend uses the frontend code to extract the frontmatter?

I think the sentence should be structured a bit differently to make clear what happens:

`All mentioned fields are extracted from the note content by <actor> on save or update, using <optional explanation of how/where>.` Actor should be "frontend" or "backend", and we should mention if they are submitted to the API or only provided by the backend through the API after extracting it.

Reply:
So if I got the addition correct, the backend uses the frontend code to extract the frontmatter?

I think the sentence should be structured a bit differently to make clear what happens:

`All mentioned fields are extracted from the note content by <actor> on save or update, using <optional explanation of how/where>.` Actor should be "frontend" or "backend", and we should mention if they are submitted to the API or only provided by the backend through the API after extracting it.

Reply:
I tried to make it more clear.

Reply:
I see, I think we should remove the "frontend code delivered by the back end" dance. We can either link to the code/a version of the code or omit this implementation detail, as it doesn't seem really relevant here.

Reply:
changed